tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -41,13 +41,42 @@ type Report struct {
 	Flaky []string `json:"flaky"`
 }
 
+// JSONClient contains the set of operations a JSON reporter needs
+type Client interface {
+	CreateReportForRepo(repo string, flaky []string, writeFile bool) (*Report, error)
+	GetFlakyTests(repo string) ([]string, error)
+	GetReportRepos() ([]string, error)
+	GetFlakyTestReport(repo string, buildID int) ([]*Report, error)
+}
+
+// Client is simply a way to call methods, it does not contain any data itself
+type JSONClient struct{}
+
 // Initialize wraps prow's init, which must be called before any other prow functions are used.
-func Initialize(serviceAccount string) error {
-	return prow.Initialize(serviceAccount)
+func Initialize(serviceAccount string) (Client, error) {
+	return NewClient(), prow.Initialize(serviceAccount)
+}
+
+// NewClient gives us a new Client struct, without initializing Prow
+func NewClient() *JSONClient {
```

Comment:
does this have to be another method or can we inline this above? Also, does this need to be public?

Reply:
I call it in the reporter since prow had already been initialized, but since the client itself just serves as a way to call methods I don't really need a constructor anyways. Fixed.